storage/kubernetes/controller.go

```diff
@@ -152,14 +153,14 @@ func (s *storage) saveInK8s(secret *v1.Secret) (*v1.Secret, error) {
 	// in favor of just blindly replacing the fields on the Kubernetes secret.
 	if s.tls != nil {
 		// merge new secret with secret from backing storage, if one exists
-		if existing, err := s.storage.Get(); err == nil && existing != nil && len(existing.Data) > 0 {
+		if existing, err := s.storage.Get(); err == nil && existing != nil && isValidTLSSecret(existing) {
```

Comment:
This call to `s.storage.Get` is protected with `RLock` in the `Get` method. Since the mutex is no longer being used to protect the call to `saveInK8s`, should we use `s.Get()` here instead?

Reply:
Good catch. I also added a wrapper for the controller check since that's a read to the base object as well.